lib/model/rwfolder.go

```diff
@@ -429,9 +429,11 @@ func (f *sendReceiveFolder) pullerIteration(ignores *ignore.Matcher) int {
 
 	for _, fi := range processDirectly {
 		// Verify that the thing we are handling lives inside a directory,
-		// and not a symlink or empty space.
-		if !osutil.IsDir(f.dir, filepath.Dir(fi.Name)) {
-			f.newError(fi.Name, errNotDir)
+		// and not a symlink or empty space. If it's already deleted, handle
+		// the check inside deleteFile deleteDir only in the scenario where
+		// the subject we are supposed to delete actually exists.
+		if !fi.IsDeleted() && !osutil.IsDir(f.dir, filepath.Dir(fi.Name)) {
```

Comment:
This seems dangerous, because a path that traverses Symlinks might end up in ``buckets``. I think this allows an attacker to delete known files outside of the folder.